src/chart_process/bms_processor.rs

```diff
@@ -759,7 +542,8 @@ impl ChartProcessor for BmsProcessor {
         let visible_window_y = self.visible_window_y();
         let scroll_factor = self.current_scroll.clone();
 
-        self.preloaded_events.iter().map(move |event_with_pos| {
+        let preloaded = self.preloaded_events.clone();
+        preloaded.into_iter().map(move |event_with_pos| {
```

Comment:
It's unnecessary to clone the vec.
```suggestion
        self.preloaded_events.iter().map(move |event_with_pos| {
```